Source/WebCore/loader/FrameLoader.cpp

```diff
@@ -1061,7 +1061,8 @@ bool FrameLoader::checkIfFormActionAllowedByCSP(const URL& url, bool didReceiveR
 
 void FrameLoader::setOpener(Frame* opener)
 {
-    if (m_opener && !opener)
+    auto* window = m_frame.window();
+    if ((m_opener && !opener) || (window && window->hasRemoteOpener()))
```

Comment:
Don't we need to let the UIProcess know when the main frame's opener changes? Since it is being stored on WebPageProxy?